frontend/src/concepts/pipelines/content/createRun/RunForm.tsx

```diff
@@ -15,65 +17,82 @@ type RunFormProps = {
   onValueChange: (key: keyof RunFormData, value: ValueOf<RunFormData>) => void;
 };
 
-const RunForm: React.FC<RunFormProps> = ({ data, onValueChange }) => (
-  <Form
-    maxWidth="500px"
-    onSubmit={(e) => {
-      e.preventDefault();
-    }}
-  >
-    <FormSection id="run-section-project-name" title="Project">
-      <FormGroup label="Project">
-        <Text>{getProjectDisplayName(data.project)}</Text>
-      </FormGroup>
-    </FormSection>
-    <FormSection
-      id={CreateRunPageSections.NAME_DESC}
-      aria-label={runPageSectionTitles[CreateRunPageSections.NAME_DESC]}
+const RunForm: React.FC<RunFormProps> = ({ data, onValueChange }) => {
+  const [latestVersion] = useLatestPipelineVersion(data.pipeline?.pipeline_id);
+  const selectedVersion = data.version || latestVersion;
+
+  const updateInputParams = React.useCallback(
+    (version: PipelineVersionKFv2 | undefined) =>
+      onValueChange(
+        'params',
+        Object.keys(version?.pipeline_spec?.root?.inputDefinitions?.parameters || {}).reduce(
+          (acc: RuntimeConfigParameters, parameter) => {
+            acc[parameter] = data.params?.[parameter] ?? '';
+            return acc;
+          },
+          {},
+        ),
+      ),
+    [data.params, onValueChange],
+  );
+
+  React.useEffect(() => {
+    if (latestVersion) {
+      onValueChange('version', latestVersion);
+      updateInputParams(latestVersion);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Help me understand what is the value of this. This eslint is important because unfortunately it is kinda dumb and will open up the next edit to not have to bide by adding a prop to this hook. 

From what I can see -- You included `onValueChange` in the previous `useCallback` & it produces the other item you have omitted here. Why not just add the deps?

Reply:
The way the form state management is setup, setting state in a useEffect that includes a non-primitive, in this case, `data.params`, there is infinite re-setting of state. I can use `useDeepCompareMemoize` to allow for exhaustive deps if you'd like.

Reply:
hmmm, I see -- okay -- leave it for now, we can address the weirdnesses of the form on another pass.